git-clone/main.tf

```
@@ -25,20 +25,82 @@ variable "agent_id" {
   type        = string
 }
 
+variable "git_providers" {
+  type = map(object({
+    provider = string
+  }))
+  description = "The set of git provider by the base url."
```

I found the wording a bit confusing, but I am not sure I can think of anything better.  What about:

```suggestion
  description = "A mapping of URLs to their git provider."
```